chia/pools/pool_wallet.py

```diff
@@ -807,13 +816,16 @@ async def claim_pool_rewards(self, fee: uint64) -> Tuple[TransactionRecord, Opti
         all_spends: List[CoinSpend] = []
         total_amount = 0
 
-        current_coin_record = None
+        first_coin_record = None
         for coin_record in unspent_coin_records:
             if coin_record.coin not in coin_to_height_farmed:
                 continue
-            current_coin_record = coin_record
-            if len(all_spends) >= 100:
-                # Limit the total number of spends, so it fits into the block
+            if first_coin_record is None:
```

Comment:
This is the bug - we were choosing the last coin we considered, but not *necessarily* the last included in the spend.

Fixed by always choosing the first.

Reply:
Thanks - I'll add a comment that explains this!

Reply:
@arvidn The reason the last coin considered might not be included is because of the line 
    `if len(all_spends) >= max_spends_in_tx`
That stops the current coin in the loop being included in `absorb_spend`, but the previous code was incorrectly using the last value of `current_coin` as if it were.